Sources/PackageGraph/Pubgrub.swift

```diff
@@ -516,6 +594,30 @@ public final class PubgrubDependencyResolver<
     /// Skip updating containers while fetching them.
     private let skipUpdate: Bool
 
+    func trace(
+        value: Traceable,
+        type: TraceStep.StepType,
+        location: TraceStep.Location,
+        cause: String?
+        ) {
+        delegate?.trace(TraceStep(value: value,
+                                  type: type,
+                                  location: location,
+                                  cause: cause,
+                                  decisionLevel: solution.decisionLevel))
+    }
+
+    func decide(_ package: Identifier, version: Version, location: TraceStep.Location) {
```

Comment:
I was thinking that we would make solution's decide API take the package and version and it would construct the term.

Reply:
You mean moving this into `PartialSolution.decide(_:)`? That would work, but I would have to construct the term for the call to `trace()` anyways unless I move the tracking of that including the calls to the delegate into the solution as well.

Reply:
Creating a term object for trace() seems fine. IMO it is better to have APIs that can avoid obvious mistakes that would otherwise be only found at runtime.